test/functional/tests/cache_ops/test_dynamic_cache_mode_switching.py

```diff
@@ -0,0 +1,321 @@
+          after stopping and reloading cache back.
+          Check also other parameters consistency after reload.
+        pass_criteria:
+          - In all cache modes data reads and writes are handled properly before and after reload.
+          - All cache parameters preserve their values after reload.
+    """
+
+    with TestRun.step("Partition cache and core devices"):
+        cache_dev, core_dev = storage_prepare()
+
+    with TestRun.step(f"Start cache in {cache_mode[0]} mode"):
+        cache = casadm.start_cache(cache_dev, cache_mode[0], force=True)
+        Udev.disable()
+
+    with TestRun.step("Add core to the cache"):
+        core = cache.add_core(core_dev)
+
+    with TestRun.step(f"Change cache mode to {cache_mode[1]}"):
+        cache.set_cache_mode(cache_mode[1], flush=True)
+        check_cache_config = cache.get_cache_config()
+
+    with TestRun.step(f"Check if {cache_mode[1]} cache mode works properly"):
+        check_cache_mode_operation(cache, core, io_size, cache_mode[1])
+
+    with TestRun.step("Stop and load cache back"):
+        cache.stop()
+        cache = casadm.load_cache(cache_dev)
+
+    with TestRun.step("Check parameters consistency"):
+        if check_cache_config == cache.get_cache_config():
+            TestRun.LOGGER.info("All parameters are consistent after reload")
+        else:
+            if check_cache_config.cache_mode != cache.get_cache_mode():
+                TestRun.fail("Cache mode do not match after reload!")
+            if check_cache_config.cleaning_policy != cache.get_cleaning_policy():
+                TestRun.fail("Cleaning policy do not match after reload!")
+            if check_cache_config.eviction_policy != cache.get_eviction_policy():
+                TestRun.fail("Eviction policy do not match after reload!")
+            if check_cache_config.cache_line_size != cache.get_cache_line_size():
+                TestRun.fail("Cache line size do not match after reload!")
+            if check_cache_config.metadata_mode != cache.get_metadata_mode():
+                TestRun.fail("Metadata mode do not match after reload!")
+
+    with TestRun.step(
+        f"Check if {cache_mode[1]} cache mode works properly after reload"
+    ):
+        if cache_mode[1] == CacheMode.WA:
+            check_wa_after_reload(cache, core, io_size)
+        else:
+            check_cache_mode_operation(cache, core, io_size, cache_mode[1])
+
+    with TestRun.step("Stop all caches"):
+        casadm.stop_all_caches()
+        Udev.enable()
+
+
+@pytest.mark.parametrize(
+    "cache_mode_1,cache_mode_2,flush",
+    [
+        (CacheMode.WT, CacheMode.WB, False),
+        (CacheMode.WB, CacheMode.PT, False),
+        (CacheMode.WB, CacheMode.PT, True),
+        (CacheMode.WB, CacheMode.WT, False),
+        (CacheMode.WB, CacheMode.WT, True),
+        (CacheMode.PT, CacheMode.WT, False),
+        (CacheMode.WT, CacheMode.WA, False),
+    ],
```

Comment:
As above - You forgot about write-only mode.

Reply:
Done